src/rasmus_fuel/realistic.py

```diff
@@ -0,0 +1,149 @@
+    u_wind: npt.ArrayLike = None,
+    v_wind: npt.ArrayLike = None,
+    w_wave_hight: npt.ArrayLike = None,
+    coeff: float = 1.0,
+    **kwargs,
+) -> npt.ArrayLike:
+    """Calculate quadratic drag law power needed to maintain speed over ground.
+    Parameters
+    ----------
+    u_ship_og: array
+        Ship eastward speed over ground in meters per second.
+    v_ship_og: array
+        Ship northward speed over ground in meters per second.
+        Needs same shape as u_ship_og.
+    course_ship_og: array
+        Ship course over ground in radians
+        needs same shape as u_ship_og
+    u_current: array
+        Ocean currents eastward speed over ground in meters per second.
+        Needs shape that can be broadcast to shape of u_ship_og and v_ship_og.
+    v_current: array
+        Ocean currents northward speed over ground in meters per second.
+        Needs shape that can be broadcast to shape of u_ship_og and v_ship_og.
+    u_wind: array
+        Eastward 10 m wind in m/s
+        Needs shape that can be broadcst to shape of u_ship_og and v_ship_og
+    v_wind: array
+        Northward 10 m wind in m/s
+        Needs shape that can be broadcst to shape of u_ship_og and v_ship_og
+    w_wave_hight: array
+        Spectral significant wave height (Hm0), meters
+        Needs shape that can be broadcst to shape of u_ship_og and v_ship_og
+    coeff: float
+        Coefficient in units of kg/m. This coefficient contains info about
+        the density of water, the effective cross section of the vessel and the
+        drag coefficient of the vessel.  Defaults to 1.0
+    All other keyword arguments will be ignored.
+    Returns
+    -------
+    array:
+        Power in W (=kg*m2/s3) needed to maintain speed over ground for each
+        element of u_ship_og and v_ship_og. Shape will be identical to
+        u_ship_og and v_ship_og.
+    """
+    # ensure shapes of u_ship_og and v_ship_og agree
+    if np.array(u_ship_og).shape != np.array(v_ship_og).shape:
+        raise ValueError('Shape of u_ship_og and v_ship_og need to agree.')
+
+    # calc speed through water
+    u_ship_tw = u_ship_og - u_current
+    v_ship_tw = v_ship_og - v_current
+
+    # calc engine power to maintain speed over ground using ocean current resistance term
+    speed_tw = (u_ship_tw ** 2 + v_ship_tw ** 2) ** 0.5
+    power_needed = coeff * (speed_tw ** 3)
+
+    # calc engine power to maintain speed over ground using : (1) ocean current resistance, (2) wind resistance, (3) wave resistance
+
+    # calc relative wind speed
+    wind_speed = (u_wind ** 2 + v_wind ** 2) ** 0.5
+    wind_dir = arctan(v_wind/u_wind)
+    wind_dir_rel = wind_dir - course_ship +  pi
+
+    u_wind_rel = wind_speed * cos(wind_dir_rel) - u_ship_og
+    v_wind_rel = wind_speed * sin(wind_dir_rel) - v_ship_og
+
+    coeff_wind_drag = 0.5* model_param["air_mass_density"]* model_param["wind_resistance_coefficient"] * model_param["vessel_supersurface_area"]
```

Comment:
The model parameter dict should be split into two:
- `vessel_param` for the vessel specifics
- `physics_param` for things like water density and gravity acceleration

Then, they should be added as keyword arguments to the function call with a fallback for default parameters if the arg is `None`.